packages/pwa-kit-react-sdk/src/ssr/universal/hooks/index.js

```diff
@@ -70,3 +71,38 @@ export const useOrigin = ({fromXForwardedHeader = false}) => {
     }
     return APP_ORIGIN
 }
+
+/**
+ * Blocks the navigation to run a provided function whenever there is a new page being navigated to
```

Comment:
This description is a little hard to understand. Do you mind taking another stab at it?